src/coreclr/jit/runtimelookup.cpp

```diff
@@ -0,0 +1,434 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include "jitpch.h"
+#ifdef _MSC_VER
+#pragma hdrstop
+#endif
+
+// Obtain constant pointer from a tree
+static void* GetConstantPointer(Compiler* comp, GenTree* tree)
+{
+    void* cns = nullptr;
+    if (tree->gtEffectiveVal()->IsCnsIntOrI())
+    {
+        cns = (void*)tree->gtEffectiveVal()->AsIntCon()->IconValue();
+    }
+    else if (comp->vnStore->IsVNConstant(tree->gtVNPair.GetLiberal()))
+    {
+        cns = (void*)comp->vnStore->CoercedConstantValue<ssize_t>(tree->gtVNPair.GetLiberal());
+    }
+    return cns;
+}
+
+// Save expression to a local and append it as the last statement in exprBlock
+static GenTree* SpillExpression(Compiler* comp, GenTree* expr, BasicBlock* exprBlock, DebugInfo& debugInfo)
+{
+    unsigned const tmpNum         = comp->lvaGrabTemp(true DEBUGARG("spilling expr"));
+    comp->lvaTable[tmpNum].lvType = expr->TypeGet();
```

Comment:
```suggestion
    comp->lvaTable[tmpNum].lvType = genActualType(expr);
```

Also would be nice to add function headers for these new functions.

Reply:
Actually you can remove this line completely -- `gtNewTempAssign` should take care to set the type properly.

Reply:
Do we do that for internal static helpers? It's like lambdas 🙂

Reply:
Removed